go-runtime/schema/analyzers/verb.go

```diff
@@ -0,0 +1,188 @@
+package analyzers
+
+import (
+	"go/ast"
+	"go/types"
+
+	"github.com/TBD54566975/ftl/backend/schema"
+	"github.com/TBD54566975/ftl/backend/schema/strcase"
+	"github.com/TBD54566975/golang-tools/go/analysis"
+	"github.com/TBD54566975/golang-tools/go/analysis/passes/inspect"
+	"github.com/TBD54566975/golang-tools/go/ast/inspector"
+	"github.com/alecthomas/types/optional"
+)
+
+// VerbExtractor extracts verbs to the module schema.
+var VerbExtractor = newExtractor("verb", extractVerbs, (*verbExtractorFact)(nil))
+
+type verbExtractorFact struct {
+	decl schema.Decl
+}
+
+func (t *verbExtractorFact) AFact() {}
+func (t *verbExtractorFact) setValue(decl schema.Decl) {
+	t.decl = decl
+}
+func (t *verbExtractorFact) getValue() schema.Decl {
+	return t.decl
+}
+
+func extractVerbs(pass *analysis.Pass) (interface{}, error) {
+	nn := NativeNames{}
+	verbs := []schema.Decl{}
+	in := pass.ResultOf[inspect.Analyzer].(*inspector.Inspector) //nolint:forcetypeassert
+	nodeFilter := []ast.Node{
+		(*ast.FuncDecl)(nil),
+	}
+	in.Preorder(nodeFilter, func(n ast.Node) {
```

Comment:
Oh `in.Preorder()` is super handy, nice nice. We can delete my shitty `astwalk` function at some point.